be/tool/beck.c

```diff
@@ -418,6 +420,7 @@ static struct builder b;
 static struct gen g[MAX_GEN] = {};
 
 static bool  dry_run = false;
+static bool  signaled = false;
 
 #define FLOG(level, s)						\
 	M0_LOG(level, "        at offset: %"PRId64" errno: %s (%i), eof: %i", \
```

Comment:
I have raised this concern in another review. But I will raise it again here.

The `errno` (and also the strerror(errno)) will report incorrect error messages, because the `errno` is not set in any Motr functions.
For example here:

```
714 static int recdo(struct scanner *s, const struct m0_format_tag *tag,
 715                  struct rectype *r)
 716 {
 717         unsigned  size = tag->ot_size + sizeof(struct m0_format_footer);
 718         void     *buf;
 719         int       result;
 720 
 721         if (size > MAX_REC_SIZE)
 722                 return M0_RC(-EINVAL);
 723 
 724         buf = alloca(size);
 725         result = get(s, buf, size);
 726         if (result == 0) {
 727                 if (memcmp(tag, &r->r_tag, sizeof *tag) == 0) {
 728                         result = m0_format_footer_verify(buf, false);
 729                         if (result != 0) {
 730                                 RLOG(M0_DEBUG, "С", s, r, tag);
 731                                 FLOG(M0_DEBUG, s);
 732                                 r->r_stats.s_chksum++;
 733                                 if (!dry_run && r->r_ops != NULL &&
 734                                     r->r_ops->ro_check != NULL)
 735                                         result = r->r_ops->ro_check(s, r, buf);
 736                         } else {
 737                                 RLOG(M0_DEBUG, "R", s, r, tag);
 738                                 if (r->r_ops != NULL &&
 739                                     r->r_ops->ro_proc != NULL)
 740                                         result = r->r_ops->ro_proc(s, r, buf);
 741                         }
 742                 } else {
 743                         RLOG(M0_DEBUG, "V", s, r, tag);
 744                         FLOG(M0_DEBUG, s);
 745                         r->r_stats.s_version++;
 746                         if (r->r_ops != NULL && r->r_ops->ro_ver != NULL)
 747                                 result = r->r_ops->ro_ver(s, r, buf);
 748                 }
 749         }
 750         return M0_RC(result);
 751 }
```
The call of FLOG() on Line 731 and Line 744 will report incorrect error messages, because it will use `errno`, but `errno` is not set in previous calls of `m0_format_footer_verify` or `get`.
The real error number 'result' returned from previous functions is ignored.

Reply:
As discussed we will check with Nikita and can be taken as separate PR.

@nikitadanilov  can you please share your views on use of 'errno' in FLOG.